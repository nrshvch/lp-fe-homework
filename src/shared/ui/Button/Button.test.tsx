import { describe, it, expect, vi, afterEach } from "vitest";
import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import { Button } from ".";

describe("<Button />", () => {
  afterEach(() => {
    cleanup();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Button>Snapshot</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders disabled state", () => {
    render(<Button disabled>Disabled</Button>);
    const btn = screen.getByRole("button");
    expect(btn.classList.contains("Button_disabled")).toBe(true);
  });

  it("fires onClick when clicked", () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick}>Test Click</Button>);
    const btn = screen.getByRole("button");
    fireEvent.click(btn);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("does not fire onClick when disabled", () => {
    const handleClick = vi.fn();
    render(
      <Button onClick={handleClick} disabled>
        Disabled
      </Button>
    );
    const btn = screen.getByRole("button");
    fireEvent.click(btn);
    expect(handleClick).not.toHaveBeenCalled();
  });
});
