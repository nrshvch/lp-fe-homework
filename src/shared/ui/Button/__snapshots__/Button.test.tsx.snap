// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button Button_default Button_md"
    type="button"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
